File: 2025-redistribute-characters-to-make-all-strings-equal/redistribute-characters-to-make-all-strings-equal.java
class Solution {
    public boolean makeEqual(String[] words) {
        int n = words.length;
        int[] map = new int[26];
        for(String word : words){
            for(char c : word.toCharArray()){
                map[c-'a']++;
            }
        }
        for(int i=0; i<26; i++){
            if(map[i]%n != 0 ) return false; 
        }
        return true; 
    }
}

File: 2025-redistribute-characters-to-make-all-strings-equal/redistribute-characters-to-make-all-strings-equal.py
class Solution:
    def makeEqual(self, words: List[str]) -> bool:
        counts = [0] * 26
        for word in words:
            for c in word:
                counts[ord(c) - ord('a')] += 1
        n = len(words)
        for c in counts:
            if c % n != 0: return False
        return True